### PR TITLE
docs(recipes): Refreshing Select Partitions

### PR DESCRIPTION
**Pull request by rchkv**
**[PREVIEW](https://deploy-preview-3498--cubejs-docs-staging.netlify.app/recipes/refreshing-select-partitions)**

I need help with the refresh key update. If you execute `docker-compose up` the first time, you can see that the order is updated, but the query received the response from the old pre-aggregation. If you restart containers, you'll see that all works as expected (the order is updated, and the partition is rebuilt). 
I tried to change `refreshKey.every` and `orchestratorOptions.queryCacheOptions.refreshKeyRenewalThreshold` to `1 second`, but this is not work. 

Any thoughts on that?

### CODE REVIEW

**Comment by adnanrahic** [comment]
@rchkv does this happen in dev mode or prod mode?

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+orders, including number and status:
+
+| id  | number | status     | created_at          | updated_at          |
+| --- | ------ | ---------- | ------------------- | ------------------- |
+| 1   | 1      | processing | 2021-08-10 14:26:40 | 2021-08-10 14:26:40 |
+| 2   | 2      | completed  | 2021-08-20 13:21:38 | 2021-08-22 13:10:38 |
+| 3   | 3      | shipped    | 2021-09-01 10:27:38 | 2021-09-02 01:12:38 |
+| 4   | 4      | completed  | 2021-09-20 10:27:38 | 2021-09-20 10:27:38 |
+
+In our case, each order has `created_at` and `update_at` properties. The
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:36]:
```suggestion
In our case, each order has `created_at` and `updated_at` properties. The
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+`update_at` property is the last order update timestamp. To create a
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:37]:
```suggestion
`updated_at` property is the last order update timestamp. To create a
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+pre-aggregation splitting to partitions, we need to specify the
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:38]:
```suggestion
pre-aggregation with partitions, we need to specify the
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+```javascript
+preAggregations: {
+    orders: {
+      type: `rollup`,
+      external: true,
+      dimensions: [CUBE.number, CUBE.status, CUBE.createdAt, CUBE.updatedAt],
+      timeDimension: CUBE.createdAt,
+      granularity: `day`,
+      partitionGranularity: `month`,
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:50]:
```suggestion
      partitionGranularity: `month`, // this is where we specify the partition
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+      refreshKey: {
+        sql: `SELECT max(updated_at) FROM public.orders`
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:52]:
```suggestion
        sql: `SELECT max(updated_at) FROM public.orders` // check for updates of the updated_at property
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md (hunk 4 above)
Comment (adnanrahic) [reply docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:50]: This is just an idea I have to let the user know exactly where in the code the partition is defined. If you have a better idea of how to write this, I'm open for suggestions. :smile:

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md (hunk 5 above)
Comment (adnanrahic) [reply docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:52]: Another suggestion, I just want the user to know exactly what this `sql` statement does. Open for suggestions on how to improve the wording. :smile:

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+      },
+    },
+  },
+```
+
+As you can see, we defined custom
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:58]:
```suggestion
As you can see, we defined custom a
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+[`refreshKey`](https://cube.dev/docs/schema/reference/pre-aggregations#parameters-refresh-key-sql)
+that will check a new value of the `updated_at` property. The refresh key is
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:60]:
```suggestion
that will check for new values of the `updated_at` property. The refresh key is
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+evaluated for each partition separately. For example, if we will update order
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:61]:
```suggestion
evaluated for each partition separately. For example, if we update orders
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+from august and update their `update_at` property, the current refresh key will
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:62]:
```suggestion
from august and update their `updated_at` property, the current refresh key will
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+  select min(("orders".created_at::timestamptz AT TIME ZONE 'UTC')) from public.orders AS "orders"
+--
+Executing SQL: 5b4c517f-b496-4c69-9503-f8cd2b4c73b6
+--
+  select max(("orders".created_at::timestamptz AT TIME ZONE 'UTC')) from public.orders AS "orders"
+--
+```
+
+Note that the query for two partitions is the same. It's the reason why **all
+partitions** will be update.
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:84]:
```suggestion
partitions** will be updated.
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+
+Well, how to fix this and update only august partition? We can use the
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:86]:
```suggestion
How do we fix this and update only the partition for august? We can use the
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+      granularity: `day`,
+      partitionGranularity: `month`,
+      refreshKey: {
+        sql: `SELECT max(updated_at) FROM public.orders WHERE ${FILTER_PARAMS.Orders.createdAt.filter('created_at')}`
+      },
+    },
+  },
+```
+
+Cube will filter data by `created_at` property and apply the refresh key for it.
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:108]:
```suggestion
Cube will filter data by the `created_at` property and then apply the refresh key for the `updated_at` property.
```

Comment (adnanrahic) [reply docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:108]: @rchkv is this okay?

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+There is how it looks in the Cube logs:
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:109]:
```suggestion
Here's how it looks in the Cube logs:
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+Executing SQL: e1155b2f-859b-4e61-a760-17af891f5f0b
+--
+  SELECT max(updated_at) FROM public.orders WHERE created_at >= '2021-08-01T00:00:00.000Z'::timestamptz AND created_at <= '2021-08-31T23:59:59.999Z'::timestamptz
+--
+Executing SQL: e1155b2f-859b-4e61-a760-17af891f5f0b
+--
+  SELECT max(updated_at) FROM public.orders WHERE created_at >= '2021-09-01T00:00:00.000Z'::timestamptz AND created_at <= '2021-09-30T23:59:59.999Z'::timestamptz
+```
+
+Note that Cube checks the refresh key value using date range over the
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:133]:
```suggestion
Note that Cube checks the refresh key value using a date range over the
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+`created_at` property. With this refresh key only one partition will be update.
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:134]:
```suggestion
`created_at` property. With this refresh key, only one partition will be updated.
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+
+## Result
+
+We have received orders from two partitions of pre-aggregation and only one of
+them is been updated when order changed their status:
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:139]:
```suggestion
them has been updated when an order changed its status:
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+--
+  SELECT max(updated_at) FROM public.orders WHERE created_at >= '2021-09-01T00:00:00.000Z'::timestamptz AND created_at <= '2021-09-30T23:59:59.999Z'::timestamptz
+```
+
+Note that Cube checks the refresh key value using date range over the
+`created_at` property. With this refresh key only one partition will be update.
+
+## Result
+
+We have received orders from two partitions of pre-aggregation and only one of
```
Comment (adnanrahic) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:138]:
```suggestion
We have received orders from two partitions of a pre-aggregation and only one of
```

**Review by adnanrahic — COMMENTED** [review]
Added a few suggestions for wording. Otherwise, it looks lovely, Anton!

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+permalink: /recipes/refreshing-select-partitions
+category: Examples & Tutorials
+subCategory: Query acceleration
+menuOrder: 6
+---
+
+## Use case
+
+We have a dataset with orders and we want to aggregate data while having decent
+performance. Orders have a created time, so we can use
```
Comment (rpaik) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:12]:
```suggestion
performance. Orders have a creation time, so we can use
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+[partitioning](https://cube.dev/docs/caching/using-pre-aggregations#partitioning)
+by time to optimize pre-aggregations build and refresh time. The problem is the
```
Comment (rpaik) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:14]:
```suggestion
by time to optimize pre-aggregations build and refresh time. The problem is that the
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+order can change status after a long time. In this case we want to rebuild only
```
Comment (rpaik) [inline docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:15]:
```suggestion
order's status can change after a long period. In this case, we want to rebuild only
```

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md (hunk 11 above)
Comment (rpaik) [reply docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:86]: We probably also want a capital 'A' for August.

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md (hunk 4 above)
Comment (rchkv) [reply docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:50]: Makes sense! I like it ☺️

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md (hunk 12 above)
Comment (rchkv) [reply docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md:108]: Yeah. it's good!

File: docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md (hunk 11 above)
Comment (rchkv) [reply docs/content/Examples-Tutorials-Recipes/Refreshing-select-partitions.md]: @rpaik exactly! Thank you!

**Comment by rchkv** [comment]
> @rchkv does this happen in dev mode or prod mode?

@adnanrahic In dev mode

**Pull request merged**